src/main/java/com/jcraft/jsch/Session.java

```diff
@@ -2716,9 +2716,13 @@ public String getServerVersion(){
     return Util.byte2str(V_S);
   }
   public String getClientVersion(){
-    return Util.byte2str(V_C);
+    return V_C == null ? jsch.getClientVersion() : Util.byte2str(V_C);
   }
   public void setClientVersion(String cv){
+    V_C = null;
```

Comment:
This will lead to an NPE if a user explicitly calls `setClientVersion(null)` because the `V_C` property is dereferenced in the `connect()` & `receive_kexinit()` methods.

Reply:
All the tests that actually show that require a Docker-container I don't have. I've changed it so that everything is using the get-method now